packages/frontend/src/utils/twoFactor/twoFactorBase.js

```diff
@@ -0,0 +1,214 @@
+import { BN } from 'bn.js';
+import * as nearApiJs from 'near-api-js';
+import { parseSeedPhrase } from 'near-seed-phrase';
+
+import {
+    ACCOUNT_HELPER_URL,
+    MULTISIG_CONTRACT_HASHES,
+    MULTISIG_MIN_AMOUNT,
+    NETWORK_ID,
+    NODE_URL,
+} from '../../config';
+import { WalletError } from '../walletError';
+
+const {
+    multisig: { Account2FA },
+    Connection,
+    keyStores: { InMemoryKeyStore },
+    KeyPair,
+} = nearApiJs;
+
+const LAK_CONVERSION_BATCH_SIZE = 50; // number of LAKs that can be converted (deleted + re-created) in one transaction
+const LAK_DISABLE_THRESHOLD = 48; // maximum number of LAKs that can be converted by the `delete` method
+
+export class TwoFactorBase extends Account2FA {
+    constructor({
+        connection,
+        accountId,
+        getCode,
+        init2fa,
+        isLedgerEnabled,
+        localStorage,
+        refreshAccount,
+        has2fa,
+    }) {
+        super(connection, accountId, {
+            storage: localStorage,
+            helperUrl: ACCOUNT_HELPER_URL,
+            getCode,
+        });
+        this.has2fa = has2fa;
+        this.init2fa = init2fa;
+        this.isLedgerEnabled = isLedgerEnabled;
+        this.refreshAccount = refreshAccount;
```

Comment:
Aside from the new methods (the last three declared in the class), the only changes to this file are where these new methods (well, function parameters) are called. This decouples the class from the redux side effects and DOM global, making the class actually testable.